Detectors/Base/src/MatLayerCylSet.cxx

```diff
@@ -177,6 +176,42 @@ void MatLayerCylSet::writeToFile(const std::string& outFName)
   outf.Close();
 }
 
+GPUd() int MatLayerCylSet::searchLayerFast(float r2, int low, int high) const
+{
+  // we can avoid the sqrt .. at the cost of more memory in the lookup
+  const auto index = int(std::sqrt(r2) * InvVoxelRDelta);
```

Comment:
code used on GPU should not use std::math_function, but CAMath::..., e.g. CAMath::Sqrt